Microsoft.Toolkit.Uwp.UI.Controls.Primitives/WrapPanel/WrapPanel.cs

```diff
@@ -166,6 +166,13 @@ protected override Size ArrangeOverride(Size finalSize)
                     foreach (var rect in row.ChildrenRects)
                     {
                         var child = Children[childIndex++];
+                        while (child.Visibility == Visibility.Collapsed)
```

Comment:
Should we be worried that we're checking for `U == 0` in measure vs. Visibility here? I imagine that would mean if we have something with no size but that's visible the index would be off, eh?

Reply:
Yes, good catch, we are removing items of 0 width/height to not add twice the margin. I should use the same check here to properly consider *visible-with-0-width* elements

Reply:
I've updated the logic to only remove collapsed items from the layout. *0-width* items will now be kept since it may be the developer choice.